internal/resourcestore/resourcestore.go

```diff
@@ -0,0 +1,157 @@
+package resourcestore
+
+import (
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const sleepTimeBeforeCleanup = 1 * time.Minute
+
+// ResourceStore is a structure that saves information about a recently created resource.
+// Resources can be added and retrieved from the store. A retrieval (Get) also removes the Resource from the store.
```

Comment:
a retrieval also removes?  That's unusual no?

Reply:
I suppose, though here we need to keep a single copy of the resource, so we're basically using move semantics. do you have a suggestion for a better method name?